hudi-utilities/pom.xml

```diff
@@ -233,6 +242,17 @@
       </exclusions>
     </dependency>
 
+    <dependency>
+      <groupId>org.apache.spark</groupId>
+      <artifactId>spark-hive_${scala.binary.version}</artifactId>
+      <exclusions>
+        <exclusion>
+          <groupId>*</groupId>
+          <artifactId>*</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
```

Comment:
Why is this dependency needed here ?

Reply:
I have the same question.  And it looks like all artifacts are excluded?

Reply:
This change seems to be made by @xushiyan 

tagging him here to see if he can provide context. original spark 3.2 default profile pr https://github.com/apache/hudi/pull/4752/files#diff-ae837be21071b3bf9746dcd837e6c93442d88d135464ed7ef871aec2e79e9c8dR205

Otherwise for now will try to remove this change since not sure if this effects anything.

Reply:
Removing this dependency seems to cause failures within the azure ci https://dev.azure.com/apache-hudi-ci-org/apache-hudi-ci/_build/results?buildId=9855&view=logs&j=dcedfe73-9485-5cc5-817a-73b61fc5dcb0&t=746585d8-b50a-55c3-26c5-517d93af9934

For now opting to keep this dependency. 
```
[ERROR] testBuildHiveSyncConfig{boolean}[1]  Time elapsed: 0.017 s  <<< ERROR!
java.lang.NoClassDefFoundError: org/apache/spark/sql/hive/HiveExternalCatalog
	at org.apache.hudi.DataSourceUtils.buildHiveSyncConfig(DataSourceUtils.java:322)
	at org.apache.hudi.TestDataSourceUtils.testBuildHiveSyncConfig(TestDataSourceUtils.java:261)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)

```